pkg/codegen/go/gen_program_test.go

```diff
@@ -104,8 +104,8 @@ func goCheck(t *testing.T, path string, _ codegen.StringSet) {
 	err = integration.RunCommand(t, "point towards local Go SDK",
 		[]string{ex, "mod", "edit",
 			fmt.Sprintf("--replace=%s=%s",
-				"github.com/pulumi/pulumi/sdk/v3/go/pulumi",
-				"../../../../../../sdk")},
+				"github.com/pulumi/pulumi/sdk/v3",
+				"../../../../../../../sdk")},
```

Comment:
Why has this changed?

Reply:
I think the Go test were running via fluke before, resolving a different "github.com/pulumi/pulumi/sdk/v3" than the local copy. Until I changed this, tests were running using the published version of our SDK, I believe? I don't know Go module resolution well enough to say, but I do know that until I changed this, tests weren't running using my local changes.

Digging into the tests, they temporarily create a `go.mod` in a subdirectory like below, then run `go mod tidy`, then add a replacement for the Pulumi SDK. But I think that last step was a no-op.

```
testdata
├── aws-optionals-pp
│   ├── go
│   │   ├── aws-optionals.go 
│   │   └── go.mod # generated at test runtime
```

From the directory containing go.mod up to `sdk` is 7 levels:

```
testdata/aws-optionals-pp/go on  AaronFriel/issue8821 [$?⇕] via  v1.17.6 
❯ ls ../../../../../../../sdk
README.md  dotnet  go  go.mod  go.sum  nodejs  proto  python
```